packages/components/src/2016-12-01/button-state.js

```diff
@@ -14,28 +14,43 @@ See the License for the specific language governing permissions and
 limitations under the License.
 
 */
+function prevState(state) {
```

Comment:
The problem stems from the fact that the state object has many (view-state) properties irrelevant to the (behavioral) state machine. 

Specifically, when in the hover or focus state, the button sets its background to `blue`. Then we pass the button's state to the state machine (including `background: 'blue'`), the state machine returns all 

Consider this chain of events:
- A button gains focus
- ...
- The button sets `background: blue` into `state`.
- The button loses focus
- The button gets its next state from the state machine
    * The button passes `state` into the state machine.
    * The state machine passes back a new object with all properties of `state` (Including `background: 'blue'`, and with `hasFocus: false`.
- Button emits a change event with next state.
- React/Angular wrapper sets each property of `state` back into the button (including `background: 'blue'`).
- Button is now configured to have a blue background.

How about we take a crack at separating the behavioral state from the view state?